staging/src/k8s.io/apiserver/pkg/storage/etcd3/store.go

```diff
@@ -714,7 +722,7 @@ func (s *store) GetList(ctx context.Context, key string, opts storage.ListOption
 
 		// take items from the response until the bucket is full, filtering as we go
 		for i, kv := range getResp.Kvs {
-			if paging && int64(v.Len()) >= pred.Limit {
+			if limitOption != nil && int64(v.Len()) >= pred.Limit {
```

Comment:
@mborsz @wojtek-t why was this change made? My understanding of this code is that either the client has requested paging, or they have not. The only case where `limitOption != nil` but `paging = false` would be in a buggy state of the code where `limit = 0`, we double it, and we have `limitOption = clientv3.WithLimit(0)`, right? Is this change (to stop using an explicit `paging` boolean) simply syntactic sugar? Can we revert the name change? IMHO this is _way_ harder to reason about and this code is complex enough as it is.

Reply:
I'm not 100% sure I understand your question.

The condition in lines 598-600 didn't change (limitOption != nil) if and only if paging was true before.
So the logic here didn't change - it's supposed to be "no-op" wrt it.

The reason for replacing it was that we now need "limitOption" (as we potentially modify it), and given that "paging bool" has actually the exact same meaning, why to have both?

Reply:
I'm not opposed - I'm just saying both are correct. It's a matter of preference (and I guess we have different preferences)
Let's continue the discussion on https://github.com/kubernetes/kubernetes/pull/109623